TrafficCapture/captureKafkaOffloader/src/main/java/org/opensearch/migrations/trafficcapture/kafkaoffloader/KafkaCaptureFactory.java

```diff
@@ -0,0 +1,88 @@
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Properties;
+import java.util.WeakHashMap;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicLong;
+
+@Slf4j
+public class KafkaCaptureFactory implements IConnectionCaptureFactory {
+
+    private static final String DEFAULT_TOPIC_NAME_FOR_TRAFFIC = "logging-traffic-topic";
+
+    private final Producer<String, byte[]> producer;
+    private final String topicNameForTraffic;
+
+    public KafkaCaptureFactory(String producerPropertiesPath, String topicNameForTraffic) throws IOException {
+        Properties producerProps = new Properties();
+        try {
+            producerProps.load(new FileReader(producerPropertiesPath));
+        } catch (IOException e) {
+            log.error("Unable to locate provided Kafka producer properties file path: " + producerPropertiesPath);
+            throw e;
+        }
+        // There is likely some default timeout/retry settings we should configure here to reduce any potential blocking
+        // i.e. the Kafka cluster is unavailable
+        producer = new KafkaProducer<>(producerProps);
+        this.topicNameForTraffic = topicNameForTraffic;
+    }
+
+    public KafkaCaptureFactory(String producerPropertiesPath) throws IOException {
+        this(producerPropertiesPath, DEFAULT_TOPIC_NAME_FOR_TRAFFIC);
+    }
+
+    @Override
+    public IChannelConnectionCaptureSerializer createOffloader(String connectionId) throws IOException {
+        AtomicLong supplierCallCounter = new AtomicLong();
+        WeakHashMap<CodedOutputStream, ByteBuffer> codedStreamToByteStreamMap = new WeakHashMap<>();
+        return new StreamChannelConnectionCaptureSerializer(connectionId,
+            () -> {
+                // Set ByteBuffer to Kafka max message size of 1MB with 1024 bytes of leeway temporarily until
+                // serializer has been updated
+                ByteBuffer bb = ByteBuffer.allocate(1024 * 1023);
+                var cos = CodedOutputStream.newInstance(bb);
+                codedStreamToByteStreamMap.put(cos, bb);
+                return cos;
+            },
+            (codedOutputStream) -> {
+                try {
+                    ByteBuffer byteBuffer = codedStreamToByteStreamMap.get(codedOutputStream);
+                    codedStreamToByteStreamMap.remove(codedOutputStream);
+                    ProducerRecord<String, byte[]> record = new ProducerRecord<>(topicNameForTraffic,
+                        String.format("%s_%d", connectionId, supplierCallCounter.incrementAndGet()),
+                        Arrays.copyOfRange(byteBuffer.array(), 0, byteBuffer.position()));
+                    // Used to essentially wrap Future returned by Producer to CompletableFuture
+                    CompletableFuture cf = new CompletableFuture<>();
+                    // Async request to Kafka cluster
+                    producer.send(record, handleProducerRecordSent(cf));
```

Comment:
I've also noticed in the Kafka documentation that the Producer deals with buffering - can you confirm that handleProducerRecordSent will fire only after the message has been ack'ed by the broker (& not just buffered, or even sent)?